src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml

```diff
@@ -18,131 +11,172 @@
   </Design.PreviewWith>
 
   <Styles.Resources>
-    <sys:Double x:Key="CalendarDatePickerCurrentDayFontSize">12</sys:Double>
+    <x:Double x:Key="CalendarDatePickerCurrentDayFontSize">12</x:Double>
+    <x:Double x:Key="CalendarDatePickerMinHeight">32</x:Double>
   </Styles.Resources>
   
   <Style Selector="CalendarDatePicker">
-
     <Setter Property="Background" Value="{DynamicResource CalendarDatePickerBackground}"/>
     <Setter Property="Foreground" Value="{DynamicResource CalendarDatePickerForeground}"/>
     <Setter Property="BorderBrush" Value="{DynamicResource CalendarDatePickerBorderBrush}"/>
     <Setter Property="BorderThickness" Value="{DynamicResource CalendarDatePickerBorderThemeThickness}"/>
     <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="MinHeight" Value="{DynamicResource CalendarDatePickerMinHeight}" />
     <Setter Property="HorizontalAlignment" Value="Left" />
     <Setter Property="VerticalAlignment" Value="Center" />
+    <Setter Property="HorizontalContentAlignment" Value="Left" />
+    <Setter Property="VerticalContentAlignment" Value="Center" />
     <Setter Property="Padding" Value="4"/>
-    
     <Setter Property="Template">
       <ControlTemplate>
         <DataValidationErrors>
-        <Grid ColumnDefinitions="*,Auto">
+          <Panel x:Name="LayoutRoot"
+                 HorizontalAlignment="Stretch"
+                 VerticalAlignment="Stretch">
+            <Border x:Name="Background"
+                    CornerRadius="{TemplateBinding CornerRadius}"
+                    BorderThickness="{TemplateBinding BorderThickness}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    Background="{TemplateBinding Background}" />
+            <Grid ColumnDefinitions="*,Auto">
+              <TextBox Name="PART_TextBox"
+                       Foreground="{TemplateBinding Foreground}"
+                       Background="Transparent"
+                       BorderBrush="Transparent"
+                       BorderThickness="0"
+                       CornerRadius="{TemplateBinding CornerRadius}"
+                       Margin="{TemplateBinding BorderThickness}"
+                       Padding="{TemplateBinding Padding}"
+                       Watermark="{TemplateBinding Watermark}"
+                       UseFloatingWatermark="{TemplateBinding UseFloatingWatermark}"
+                       HorizontalAlignment="Stretch"
+                       VerticalAlignment="Stretch"
+                       VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}"
+                       HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
+                       Grid.Column="0" />
+              <Button Name="PART_Button"
+                      Grid.Column="1"
+                      Focusable="False" />
```

Comment:
This button is not focusable, and I don't see how user can open flyout.
That's probably old behavior, but I would expect "Alt+ArrowDown" to work here, as it works in SplitButton. Might be done separately.

Reply:
I can add that functionality without too much trouble I think. Will be next week before I circle back to commits though.

Reply:
Done to open the Popup/Flyout only. Escape is required to close it. This is because the Calendar handles the down key and it just creates a bad user experience. Further details in comments.